meterfs/common.c

```diff
@@ -0,0 +1,60 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Meterfs tests common part
+ *
+ * Copyright 2021 Phoenix Systems
+ * Author: Tomasz Korniluk
+ *
+ *
+ * %LICENSE%
+ */
+
+#include "common.h"
+
+static struct {
+	char buff[40];
+}common;
+
+void common_fileInfoFill(file_info_t *info, size_t sectors, size_t filesz, size_t recordsz, size_t recordcnt)
+{
+	info->filesz = filesz;
+	info->sectors = sectors;
+	info->recordsz = recordsz;
+	info->recordcnt = recordcnt;
+}
+
+
+void common_preallocOpenFile(const char *name, size_t sectors, size_t filesz, size_t recordsz, int *fd)
+{
+	TEST_ASSERT_EQUAL(0, file_allocate(name, sectors, filesz, recordsz));
+	snprintf(common.buff, sizeof(common.buff), "/%s", name);
+	*fd = file_open(common.buff);
+	TEST_ASSERT_GREATER_OR_EQUAL(0, *fd);
+
+	memset(common.buff, 0, sizeof(common.buff));
+}
+
+
+void common_readContent(int fd, size_t offset, void *buff, size_t bufflen, void *content, int contentsz, const char *msg)
+{
+	int i;
+
+	memcpy(common.buff, content, contentsz);
+	TEST_ASSERT_EQUAL_MESSAGE(bufflen, file_read(fd, offset, buff, bufflen), msg);
```

Comment:
Just want to be sure – does unity checks for NULL msg?

Reply:
Yes unity checks that.